code/game/atoms_movable.dm

```diff
@@ -18,7 +18,16 @@
 
 //===========================================================================
 /atom/movable/Dispose()
-	for(var/atom/movable/I in contents) cdel(I)
+	for(var/atom/movable/I in contents)
+		if(istype(I,/obj/item/device/radio/detpack)) //Exception for detpacks that shouldn't be deleted, such as from when a roller bed folds
+			var/obj/item/device/radio/detpack/D = I
+			if(D.plant_target == src)
+				D.loc = get_turf(src)
+				D.nullvars()
+			else
+				cdel(I)
+		else
+			cdel(I)
```

Comment:
You should avoid this pattern of elses when you can make it more clear by doing
```
if(!istype(I, /obj/item/device/radio/detpack))
  cdel(I)
  continue
```
The cdel is the atypical event so you should handle it separately before moving on to the more standard functionality